### ISSUE TITLE
Add Linux cosim regression test

### ISSUE THREAD

**Issue by dpetrisko** [black-parrot/black-parrot#399]
Linux boot is roughly 265M instructions to the login prompt.  This means we could split it into roughly 30 10m instruction checkpoints and run in parallel to have a ~6 hour linux regression.  We can add this to our dev CI branch to run nightly (or have people run locally).

TODO:
- [x] Add run N instruction capability to testbench (most likely in RTL)
- [x] Add generate K checkpoints of size N capability to makefiles
- [x] Write regression script which generates the checkpoints and runs them in parallel
